tests/pamiq_core/threads/test_base.py

```diff
@@ -120,3 +128,165 @@ def on_tick(self) -> None:
+    """A valid BackgroundThread subclass for testing purposes."""
+
+    THREAD_TYPE = ThreadTypes.INFERENCE
+
+    def __init__(self) -> None:
+        super().__init__()
+
+
+class TestBackgroundThread:
+    @pytest.fixture
+    def background_thread(self) -> ValidBackgroundThread:
+        """Fixture to create a valid BackgroundThread instance."""
+        return ValidBackgroundThread()
+
+    @pytest.fixture
+    def thread_controller(self) -> ThreadController:
+        """Fixture to create a thread controller."""
+        return ThreadController()
+
+    @pytest.fixture
+    def read_only_controller(self, thread_controller) -> ReadOnlyController:
+        """Fixture to create a read-only controller."""
+        return ReadOnlyController(thread_controller)
+
+    @pytest.fixture
+    def controller_command_handler(
+        self, read_only_controller
+    ) -> ControllerCommandHandler:
+        """Fixture to create a ControllerCommandHandler instance."""
+        return ControllerCommandHandler(read_only_controller)
+
+    @pytest.fixture
+    def background_thread_with_controller(
+        self, background_thread, read_only_controller
+    ) -> ValidBackgroundThread:
+        """Fixture to create a BackgroundThread with a controller."""
+        background_thread.attach_controller(read_only_controller)
+        return background_thread
+
+    def test_init_with_THREAD_TYPE_attribute_other_than_control(self) -> None:
+        """Test that the Thread class can be initialized with a THREAD_TYPE
+        attribute other than 'control'."""
+
+        class TestThreadWithAppropriateThreadType(BackgroundThread):
+            THREAD_TYPE = ThreadTypes.INFERENCE
+
+            def __init__(self) -> None:
+                super().__init__()
+
+        _ = TestThreadWithAppropriateThreadType()  # no error should be raised
+
+    def test_init_with_control_THREAD_TYPE(self) -> None:
+        """Test that the Thread class can NOT be initialized with a 'control'
+        THREAD_TYPE attribute."""
+
+        class TestThreadWithControlThreadType(BackgroundThread):
+            THREAD_TYPE = ThreadTypes.CONTROL
+
+            def __init__(self) -> None:
+                super().__init__()
+
+        with pytest.raises(
+            ValueError,
+            match="BackgroundThread cannot be of type 'control'.",
+        ):
+            _ = TestThreadWithControlThreadType()
+
+    def test_thread_status(self, background_thread) -> None:
+        """Test that the thread status returns ReadOnlyThreadStatus
+        instance."""
+        assert isinstance(background_thread.thread_status, ReadOnlyThreadStatus)
+
+    def test_attach_controller(self, background_thread, read_only_controller) -> None:
+        """Test that the controller can be attached to the thread."""
+
+        background_thread.attach_controller(read_only_controller)
+        assert isinstance(
+            background_thread._controller_command_handler, ControllerCommandHandler
+        )
```

Comment:
イベントが渡されていることのテストはこのようにすると良いと思います。
```suggestion
        handler = background_thread._controller_command_handler
        assert isinstance(handler, ControllerCommandHandler)
        assert handler.on_paused == background_thread.on_paused
        assert handler.on_resumed == background_thread.on_resumed
```